recipes/cyclonedds/all/conanfile.py

```diff
@@ -0,0 +1,190 @@
+import os
+import textwrap
+from conan import ConanFile
+from conans import CMake
+from conan.errors import ConanInvalidConfiguration
+from conan.tools import files, build, scm
+
+required_conan_version = ">=1.43.0"
+
+
+class CycloneDDSConan(ConanFile):
+    name = "cyclonedds"
+    license = "EPL-2.0"
+    homepage = "https://cyclonedds.io/"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Eclipse Cyclone DDS - An implementation"\
+                  " of the OMG Data Distribution Service (DDS) specification"
+    topics = ("dds", "ipc", "ros", "middleware")
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "ssl": [True, False],
+        "shm" : [True, False],
+        "security" : [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "ssl": False,
+        "shm": False,
+        "security": False
+    }
+
+    generators = ["cmake", "cmake_find_package_multi"]
+    _cmake = None
+    short_paths = True
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join("lib", "cmake", f"conan-official-{self.name}-targets.cmake")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _MS_runtime_files(self):
+        return ["concrt140.dll",
+                "msvcp140.dll",
+                "msvcp140_1.dll",
+                "msvcp140_2.dll",
+                "msvcp140_atomic_wait.dll",
+                "msvcp140_codecvt_ids.dll",
+                "vcruntime140.dll",
+                "vcruntime140_1.dll"
+                ]
+
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.shm:
+            self.requires("iceoryx/2.0.0")
+        if self.options.ssl:
+            self.requires("openssl/1.1.1q")
+        self.tool_requires("cmake/3.16.2")
```

Comment:
```suggestion

    def build_requirements(self):
        self.tool_requires("cmake/3.16.2")
```
https://docs.conan.io/en/latest/devtools/build_requires.html#declaring-tool-requirements

Reply:
thanks! i will add